src/NumericalAlgorithms/LinearOperators/PartialDerivatives.cpp

```diff
@@ -98,42 +100,70 @@ void logical_partial_derivative(
 
   set_number_of_grid_points(logical_derivative_of_u,
                             mesh.number_of_grid_points());
-  const Matrix empty_matrix{};
-  std::array<std::reference_wrapper<const Matrix>, Dim> diff_matrices{
-      make_array<Dim, std::reference_wrapper<const Matrix>>(empty_matrix)};
-  for (size_t d = 0; d < Dim; ++d) {
-    gsl::at(diff_matrices, d) =
-        std::cref(Spectral::differentiation_matrix(mesh.slice_through(d)));
-  }
+  if (Dim == 3 and mesh.basis(1) == Spectral::Basis::SphericalHarmonic) {
+    if constexpr (std::is_same_v<typename DataType::value_type, double>) {
+      const Matrix& differentiation_matrix_xi =
+          Spectral::differentiation_matrix(mesh.slice_through(0));
+      const auto& ylm = ylm::get_spherepack_cache(mesh.extents(1) - 1);
+      for (size_t storage_index = 0; storage_index < u.size();
+           ++storage_index) {
+        const auto u_tensor_index = u.get_tensor_index(storage_index);
+        partial_derivatives_detail::apply_matrix_in_first_dim(
+            // NOLINTNEXTLINE(readability-redundant-smartptr-get)
+            logical_derivative_of_u->get(prepend(u_tensor_index, 0_st)).data(),
+            u[storage_index].data(), differentiation_matrix_xi,
+            num_grid_points);
+        const auto du = std::array{
+            // NOLINTNEXTLINE(readability-redundant-smartptr-get)
+            logical_derivative_of_u->get(prepend(u_tensor_index, 1_st)).data(),
+            // NOLINTNEXTLINE(readability-redundant-smartptr-get)
+            logical_derivative_of_u->get(prepend(u_tensor_index, 2_st)).data()};
+        ylm.gradient_all_offsets(du, make_not_null(u[storage_index].data()),
+                                 mesh.extents(0));
+      }
+    } else {
+      ERROR("Cannot use spherical harmonics with complex numbers");
```

Comment:
Maybe `Support for complex numbers with spherical harmonics is not yet implemented for logical_partial_derivative.`?